From 25b38dee43559d58f1bf9cc464107f32f476c753 Mon Sep 17 00:00:00 2001
Message-Id: <25b38dee43559d58f1bf9cc464107f32f476c753.1616318066.git.stefan@agner.ch>
In-Reply-To: <abcac570bfe1db867fda504942e6c782ada2aa00.1616318066.git.stefan@agner.ch>
References: <abcac570bfe1db867fda504942e6c782ada2aa00.1616318066.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Fri, 19 Mar 2021 15:54:30 +0100
Subject: [PATCH] Avoid autoreconf error if introspection macros are not
 available

Don't make macros for introspection mandatory. This allows to run
autoreconf even without GObject introspection automake macros available.

Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 configure.ac | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 7d95bc91..30096af9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -123,7 +123,10 @@ AC_ARG_ENABLE(daemon,
               enable_daemon=yes)
 AM_CONDITIONAL(ENABLE_DAEMON, test "$enable_daemon" != no)
 
-GOBJECT_INTROSPECTION_CHECK([0.6.2])
+m4_ifdef([GOBJECT_INTROSPECTION_CHECK], [
+    GOBJECT_INTROSPECTION_CHECK([0.6.2])
+])
+AM_CONDITIONAL(HAVE_INTROSPECTION, test "x$found_introspection" = "xyes")
 
 # Behavior
 #
-- 
2.30.2

